hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -262,6 +262,12 @@ protected Data removeInternal(Data key) {
     }
 
     @Override
+    protected void removeAllInternal(Predicate predicate) {
+        super.removeAllInternal(predicate);
+        nearCache.clear();
```

Comment:
Maybe a silly suggestion but what about this (of course this needs some refactoring): 

```
    @Override
    protected void removeAllInternal(Predicate predicate) {
        final AbstractEntryProcessor evictingProcessor = new AbstractEntryProcessor() {
            @Override
            public Object process(Entry entry) {
                nearCache.remove(toData(entry.getKey(), partitionStrategy));
                ((LazyMapEntry) entry).remove();
                return null;
            }
        };
        
        OperationFactory operation = operationProvider.createPartitionWideEntryWithPredicateOperationFactory(name,
                evictingProcessor, predicate);
        try {
            operationService.invokeOnAllPartitions(SERVICE_NAME, operation);
        } catch (Throwable t) {
            throw rethrow(t);
        }
    }
```